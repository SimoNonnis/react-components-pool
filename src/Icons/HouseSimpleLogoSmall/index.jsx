import React from 'react';
import PropTypes from 'prop-types';

/**
 * HouseSimpleLogoSmall Icon.
 * @param {number} width
 * @param {number} height
 * @param {string} fill
 * @param {string} className
 * @param {string} classNamePath
 * @returns {React.Component}
 */
function HouseSimpleLogoSmall({ width, height, fill, className, classNamePath }) {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      viewBox="0 0 78.1 139"
      width={width}
      height={height}
      className={className}
    >
      <title>HouseSimpleLogoSmall Small Icon</title>
      <path
        fill={fill}
        className={classNamePath}
        d="M3.6,80.6H0v56.3A27.3,27.3,0,0,0,9.4,139a33,33,0,0,0,10-2.1V89.8C19.4,89.2,13.6,80.6,3.6,80.6Z"
      />
      <path
        fill={fill}
        className={classNamePath}
        d="M74.4,67.3,23.6,43a8.09,8.09,0,0,1-3.2-2.7,10.71,10.71,0,0,1-1-4.5V11.9C16.2,1.7,2.6.1,2.6.1.9.1,0,0,0,0V47.7c0,2,.2,3.4,1.1,4.6a12.87,12.87,0,0,0,3,2.4L53.4,77.9s3.3,1.5,4.4,3.4.9,5.1.9,5.1v50.5a23.89,23.89,0,0,0,9,2.1c4.6,0,10.4-2.1,10.4-2.1V73.1A6.29,6.29,0,0,0,74.4,67.3Z"
      />
    </svg>
  );
}

HouseSimpleLogoSmall.propTypes = {
  width: PropTypes.number,
  height: PropTypes.number,
  fill: PropTypes.string,
  className: PropTypes.string,
  classNamePath: PropTypes.string,
};

HouseSimpleLogoSmall.defaultProps = {
  width: 20,
  height: 36,
  fill: '#1E2938',
  className: 'HouseSimpleLogoSmallSmall',
  classNamePath: 'HouseSimpleLogoSmallSmallPath',
};

export default HouseSimpleLogoSmall;
