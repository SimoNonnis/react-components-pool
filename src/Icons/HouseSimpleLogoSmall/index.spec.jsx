/* eslint-env jest */
import React from 'react';

import HouseSimpleLogoSmall from './';

describe('Test <HouseSimpleLogoSmall /> icon', () => {
  it('should render HouseSimpleLogoSmall icon correctly with default props', () => {
    const wrapper = shallow(<HouseSimpleLogoSmall />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render HouseSimpleLogoSmall icon correctly with props', () => {
    const wrapper = shallow(
      <HouseSimpleLogoSmall fill="#a4a1a1" className="pof" height={30} width={30} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
