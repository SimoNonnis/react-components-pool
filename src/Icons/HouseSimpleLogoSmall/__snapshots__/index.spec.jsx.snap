// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <HouseSimpleLogoSmall /> icon should render HouseSimpleLogoSmall icon correctly with default props 1`] = `
<svg
  className="HouseSimpleLogoSmallSmall"
  height={36}
  viewBox="0 0 78.1 139"
  width={20}
  xmlns="http://www.w3.org/2000/svg"
>
  <title>
    HouseSimpleLogoSmall Small Icon
  </title>
  <path
    className="HouseSimpleLogoSmallSmallPath"
    d="M3.6,80.6H0v56.3A27.3,27.3,0,0,0,9.4,139a33,33,0,0,0,10-2.1V89.8C19.4,89.2,13.6,80.6,3.6,80.6Z"
    fill="#1E2938"
  />
  <path
    className="HouseSimpleLogoSmallSmallPath"
    d="M74.4,67.3,23.6,43a8.09,8.09,0,0,1-3.2-2.7,10.71,10.71,0,0,1-1-4.5V11.9C16.2,1.7,2.6.1,2.6.1.9.1,0,0,0,0V47.7c0,2,.2,3.4,1.1,4.6a12.87,12.87,0,0,0,3,2.4L53.4,77.9s3.3,1.5,4.4,3.4.9,5.1.9,5.1v50.5a23.89,23.89,0,0,0,9,2.1c4.6,0,10.4-2.1,10.4-2.1V73.1A6.29,6.29,0,0,0,74.4,67.3Z"
    fill="#1E2938"
  />
</svg>
`;

exports[`Test <HouseSimpleLogoSmall /> icon should render HouseSimpleLogoSmall icon correctly with props 1`] = `
<svg
  className="pof"
  height={30}
  viewBox="0 0 78.1 139"
  width={30}
  xmlns="http://www.w3.org/2000/svg"
>
  <title>
    HouseSimpleLogoSmall Small Icon
  </title>
  <path
    className="HouseSimpleLogoSmallSmallPath"
    d="M3.6,80.6H0v56.3A27.3,27.3,0,0,0,9.4,139a33,33,0,0,0,10-2.1V89.8C19.4,89.2,13.6,80.6,3.6,80.6Z"
    fill="#a4a1a1"
  />
  <path
    className="HouseSimpleLogoSmallSmallPath"
    d="M74.4,67.3,23.6,43a8.09,8.09,0,0,1-3.2-2.7,10.71,10.71,0,0,1-1-4.5V11.9C16.2,1.7,2.6.1,2.6.1.9.1,0,0,0,0V47.7c0,2,.2,3.4,1.1,4.6a12.87,12.87,0,0,0,3,2.4L53.4,77.9s3.3,1.5,4.4,3.4.9,5.1.9,5.1v50.5a23.89,23.89,0,0,0,9,2.1c4.6,0,10.4-2.1,10.4-2.1V73.1A6.29,6.29,0,0,0,74.4,67.3Z"
    fill="#a4a1a1"
  />
</svg>
`;
